src/test/java/be/seeseemelk/mockbukkit/ServerMockTest.java

```diff
@@ -544,6 +543,72 @@ public void testDefaultPotionEffects()
 		}
 	}
 
+	@Test
+	public void testYamlSerialization()
+	{
+		final String SERVER_SETTING_FILE = "bukkit.yml";
+		File file = null;
+		try {
+			Path tempFile = Files.createTempFile(SERVER_SETTING_FILE, String.valueOf(Math.random()));
+			file = new File(tempFile.toUri());
+			file.deleteOnExit();
+		}
+		catch (Exception e)
+		{
+			fail(e.getMessage());
+		}
+
+		ServerSettingsMock serverSettings = null;
+		ServerSettingsMock dummy = new ServerSettingsMock();
+
+		//test on clean start
+		try
+		{
+			serverSettings = server.getServerSettings();
+		}
+		catch (Exception e)
+		{
+			fail(e.getMessage());
+		}
+		assert serverSettings != null;
+		assertEquals(serverSettings, dummy);
+
+		//test on changing props
+		ServerSettingsMock dummy2 = new ServerSettingsMock();
```

Comment:
Can this not be done in a separate test? Testing both these states in a single test makes this test a little busy.